### ISSUE TITLE
Engine creation fails when connecting to Postgres database with an override on version()

### ISSUE THREAD

**Issue by arredond** [sqlalchemy/sqlalchemy#6912]
### Describe the bug

I'm trying to connect to a Postgres database with a custom search path that overrides the internal PG functions in the `pg_catalog` schema. However, the connection fails because SQLAlchemy makes a plain `SELECT version()` call to find out the server version, without specifying the schema that the `version()` function should be in.

This could be fixed by simply qualifying [the `version` call](https://github.com/sqlalchemy/sqlalchemy/blob/b64ecb03a5411dd5f32e40ac564bec9a886d3672/lib/sqlalchemy/dialects/postgresql/base.py#L3427) to `SELECT pg_catalog.version()`. I'm happy to provide a PR for this.

### To Reproduce

```python
The DB must contain a schema with its own `version()` function:


CREATE SCHEMA IF NOT EXISTS testing_schema;
SET search_path = testing_schema,pg_catalog,public;

-- This function will be created in the testing_schema schema
CREATE OR REPLACE FUNCTION VERSION
() 
RETURNS TEXT 
AS $$
    SELECT '1.0.0'
$$ LANGUAGE SQL IMMUTABLE;


Assume that the DB contains a schema defined in the environment variable `PG_SCHEMA` which holds its own `version()` function:

```python
import os

from sqlalchemy import create_engine

PG_CONFIG = {
    'host': os.environ['PG_HOST'],
    'database': os.environ['PG_DATABASE'],
    'user': os.environ['PG_USER'],
    'password': os.environ['PG_PASSWORD'],
    'port': 5432,
}
CONNECT_OPTIONS = f"-c search_path=testing_schema,pg_catalog,public"

connection_string = 'postgresql://{user}:{password}@{host}:{port}/{database}'.format(**PG_CONFIG)
engine = create_engine(connection_string, connect_args={'options': CONNECT_OPTIONS})
engine.connect()
```

### Error
```
AssertionError                            Traceback (most recent call last)
<ipython-input-7-4070ba9f91a5> in <module>
----> 1 engine.connect()

~/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py in connect(self, close_with_result)
   3164         """
   3165 
-> 3166         return self._connection_cls(self, close_with_result=close_with_result)
   3167 
   3168     @util.deprecated(

~/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py in __init__(self, engine, connection, close_with_result, _branch_from, _execution_options, _dispatch, _has_events, _allow_revalidate)
     94                 connection
     95                 if connection is not None
---> 96                 else engine.raw_connection()
     97             )
     98 

~/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py in raw_connection(self, _connection)
   3243 
   3244         """
-> 3245         return self._wrap_pool_connect(self.pool.connect, _connection)
   3246 
   3247 

~/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py in _wrap_pool_connect(self, fn, connection)
   3210         dialect = self.dialect
   3211         try:
-> 3212             return fn()
   3213         except dialect.dbapi.Error as e:
   3214             if connection is None:

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/base.py in connect(self)
    305 
    306         """
--> 307         return _ConnectionFairy._checkout(self)
    308 
    309     def _return_conn(self, record):

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/base.py in _checkout(cls, pool, threadconns, fairy)
    765     def _checkout(cls, pool, threadconns=None, fairy=None):
    766         if not fairy:
--> 767             fairy = _ConnectionRecord.checkout(pool)
    768 
    769             fairy._pool = pool

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/base.py in checkout(cls, pool)
    423     @classmethod
    424     def checkout(cls, pool):
--> 425         rec = pool._do_get()
    426         try:
    427             dbapi_connection = rec.get_connection()

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/impl.py in _do_get(self)
    144             except:
    145                 with util.safe_reraise():
--> 146                     self._dec_overflow()
    147         else:
    148             return self._do_get()

~/.local/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py in __exit__(self, type_, value, traceback)
     68             self._exc_info = None  # remove potential circular references
     69             if not self.warn_only:
---> 70                 compat.raise_(
     71                     exc_value,
     72                     with_traceback=exc_tb,

~/.local/lib/python3.8/site-packages/sqlalchemy/util/compat.py in raise_(***failed resolving arguments***)
    205 
    206         try:
--> 207             raise exception
    208         finally:
    209             # credit to

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/impl.py in _do_get(self)
    141         if self._inc_overflow():
    142             try:
--> 143                 return self._create_connection()
    144             except:
    145                 with util.safe_reraise():

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/base.py in _create_connection(self)
    251         """Called by subclasses to create a new ConnectionRecord."""
    252 
--> 253         return _ConnectionRecord(self)
    254 
    255     def _invalidate(self, connection, exception=None, _checkin=True):

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/base.py in __init__(self, pool, connect)
    366         self.__pool = pool
    367         if connect:
--> 368             self.__connect()
    369         self.finalize_callback = deque()
    370 

~/.local/lib/python3.8/site-packages/sqlalchemy/pool/base.py in __connect(self)
    620             # init of the dialect now takes place within the connect
    621             # event, so ensure a mutex is used on the first run
--> 622             pool.dispatch.connect.for_modify(
    623                 pool.dispatch
    624             )._exec_w_sync_on_first_run(self.connection, self)

~/.local/lib/python3.8/site-packages/sqlalchemy/event/attr.py in _exec_w_sync_on_first_run(self, *args, **kw)
    327             with self._exec_once_mutex:
    328                 try:
--> 329                     self(*args, **kw)
    330                 except:
    331                     raise

~/.local/lib/python3.8/site-packages/sqlalchemy/event/attr.py in __call__(self, *args, **kw)
    341             fn(*args, **kw)
    342         for fn in self.listeners:
--> 343             fn(*args, **kw)
    344 
    345     def __len__(self):

~/.local/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py in go(*arg, **kw)
   1689             once_fn = once.pop()
   1690             try:
-> 1691                 return once_fn(*arg, **kw)
   1692             except:
   1693                 if retry_on_exception:

~/.local/lib/python3.8/site-packages/sqlalchemy/engine/create.py in first_connect(dbapi_connection, connection_record)
    672 
    673             try:
--> 674                 dialect.initialize(c)
    675             finally:
    676                 # note that "invalidated" and "closed" are mutually

~/.local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py in initialize(self, connection)
    773 
    774     def initialize(self, connection):
--> 775         super(PGDialect_psycopg2, self).initialize(connection)
    776         self._has_native_hstore = (
    777             self.use_native_hstore

~/.local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/base.py in initialize(self, connection)
   3180 
   3181     def initialize(self, connection):
-> 3182         super(PGDialect, self).initialize(connection)
   3183 
   3184         if self.server_version_info <= (8, 2):

~/.local/lib/python3.8/site-packages/sqlalchemy/engine/default.py in initialize(self, connection)
    392     def initialize(self, connection):
    393         try:
--> 394             self.server_version_info = self._get_server_version_info(
    395                 connection
    396             )

~/.local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/base.py in _get_server_version_info(self, connection)
   3433         )
   3434         if not m:
-> 3435             raise AssertionError(
   3436                 "Could not determine version from string '%s'" % v
   3437             )

AssertionError: Could not determine version from string '1.0.0'
```
### Versions

- OS: Ubuntu 20.04
- Python: 3.8.10
- SQLAlchemy: 1.4.23
- Database: PostgreSQL 11.9
- DBAPI: psycopg2 2.8.4 (dt dec pq3 ext lo64)

### Additional context

This could be fixed by simply qualifying [the `version` call](https://github.com/sqlalchemy/sqlalchemy/blob/b64ecb03a5411dd5f32e40ac564bec9a886d3672/lib/sqlalchemy/dialects/postgresql/base.py#L3427) to `SELECT pg_catalog.version()`. I'm happy to provide a PR for this.

**Comment by CaselIT** [comment]
Hi,

This is a somewhat unusual use case, but I don't think qualifying the version call will create any unforeseen issue, so it would be nice if you could provide a PR

**Comment by zzzeek** [comment]
please send a PR, thanks

**Comment by arredond** [comment]
Just did :) Let me know if there's anything else I can do

**Comment by sqla-tester** [comment]
**arredond** has proposed a fix for this issue in the **master** branch:

**Fixes: #6912 - Postgres qualify server version call** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3015

**Comment by sqla-tester** [comment]
**arredond** has proposed a fix for this issue in the **master** branch:

**Qualify server version call in PostgreSQL** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3015
